### ISSUE TITLE
Users on production logged out of the system

### ISSUE THREAD

**Issue by AnnieMungai** [opensrp/opensrp-client-giz-malawi#423]
- [x] Randomly users are being logged out at different levels in the system. I noted that login was successful at third attempt; first attempt – system printed wrong credentials; the second attempt – you have been logged out then third attempt user credentials accepted. Not sure if this was observed throughout but I observed this through demo.
- [ ] All users logged into the tablet as well were simultaneously logged out (an update: Changed my internet to EDGE, device specs - 2GBRAM and 64GB memory - using these specs, I was logged out with the error at some point. Logging back into the app completed the sync on preview instance).
- [ ] Users already logged in are logged put at the same time and all get the message: user account has been disabled. 
![image](https://user-images.githubusercontent.com/58034047/128363218-34b9f797-25bb-4343-9f1a-b6ebbd1b6f69.png)

**Comment by AnnieMungai** [comment]
Addtional note:

**Comment by paulinembabu** [comment]
@AnnieMungai 
So the issue is caused by use of different cases of usernames  when logging in . Openmrs returns the same user even if different cases are used but when authentication process finishes in openSRP it generates same authentication_id (which should be unique) with the response from openmrs since the response will be the same for different cases of the username.

**Comment by AnnieMungai** [comment]
@paulinembabu how do we resolve this?

**Comment by paulinembabu** [comment]
@AnnieMungai I am looking at the code to determine where the issue is, then I will go ahead and come up with a proper fix for this. Also, if @ndegwamartin  is able to confirm if this is the same issue that they had, then it can speed up the Process.

**Comment by ndegwamartin** [comment]
Looks like the app differentiates between a username with lowercase from one with uppercase which means **provider1** and **PROVIDER1** are not the same during offline authentication. 

I think this can be fixed on the client-core library to enforce case insensitivity

**Comment by paulinembabu** [comment]
@ndegwamartin But the same still happens even when the app is online

**Comment by ndegwamartin** [comment]
> @ndegwamartin But the same still happens even when the app is online

Sorry, let me rephrase my statement. I meant that it happens when we already have the user having offline authentication details persisted and we try to login in again with a different case username. This forces a remote login while online

**Comment by paulinembabu** [comment]
@ndegwamartin  Okay,  that makes sense now

**Comment by AnnieMungai** [comment]
Closing this for now as it is fixed. 

Users all logged out at the same time will require a separate issue.

**Comment by bennsimon** [comment]
@paulinembabu I think this is related https://sentry.k8s.onalabs.org/organizations/ona/issues/3047/?query=&statsPeriod=14d.
